src/Entity/Character.cpp

```diff
@@ -54,34 +55,51 @@ void Character::CharacterUpdate(flecs::entity characterEntity, TransformComponen
 	animationStateComponent.mCurrentAnimName = AnimationName::RUN;
 	
 	//Changing orientation of the character depending on where it's moving
-	direction.x < 0.f ? transformComponent.mScale.x = -1.f : transformComponent.mScale.x = 1.f;
+	direction.x < 0.f ? transformComponent.mScale.x = -fabs(transformComponent.mScale.x) : transformComponent.mScale.x = fabs(transformComponent.mScale.x);
 	
 	//Check for map bounds
-	flecs::entity mapEntity = characterEntity.world().component<MapTag>().target<MapTag>();
-	Texture2D mapTexture = mapEntity.get_ref<TextureComponent>()->mTexture;
+	const flecs::entity mapEntity = characterEntity.world().component<MapTag>().target<MapTag>();
+	const Texture2D mapTexture = mapEntity.get_ref<TextureComponent>()->mTexture;
 
-	// Calculate new position
-	Vector2 newPosition = Vector2Add(transformComponent.mPosition, Vector2Scale(Vector2Normalize(direction), CHARACTER_SPEED * GetFrameTime()));
+	const Vector2 newPosition = Vector2Add(transformComponent.mPosition, Vector2Scale(Vector2Normalize(direction), CHARACTER_SPEED * GetFrameTime()));
 
 	// Check if the new position is within map boundaries 
-	//QUESTION: if i want this as function, would it be a system on map? how do i then call one?
+	//TODO: Write a function to do it in map namespace? if i find myself using same code for towers/enemies
 	if (newPosition.x >= 0 && newPosition.x <= mapTexture.width-CHARACTER_WIDTH &&
 		newPosition.y >= 0 && newPosition.y <= mapTexture.height-CHARACTER_HEIGHT)
 	{
-		  // Create a new rectangle that represents where the character would move
-		Rectangle newCharacterRect = {newPosition.x, newPosition.y, collisionComponent.mCollisionRect.width, collisionComponent.mCollisionRect.height};
+		const Rectangle newCharacterRect {newPosition.x, newPosition.y, collisionComponent.mRectScale.x, collisionComponent.mRectScale.y};
```

Comment:
When initializing directly, no need for the space between the variable name and `{`. `hasCollided` below is a good example.